<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowBase</class>
 <widget class="QMainWindow" name="MainWindowBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>651</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>750</width>
    <height>550</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Synergy</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>20</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="m_layout1">
      <item>
       <widget class="QLabel" name="m_pLabelComputerName">
        <property name="text">
         <string notr="true">This computer's name:</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="m_pSpacerUpdate">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="m_pLabelUpdate">
        <property name="styleSheet">
         <string notr="true">padding: 3px 5px;
border-radius: 3px;
background-color: #3B67D3;
color: #ffffff;
top: 3px;</string>
        </property>
        <property name="text">
         <string notr="true">m_pLabelUpdate</string>
        </property>
        <property name="openExternalLinks">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <spacer name="verticalSpacer_5">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::Policy::Fixed</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>10</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QLabel" name="m_pLabelIpAddresses">
      <property name="text">
       <string notr="true">This computer's IP addresses:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="m_pMainWidgets" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <property name="spacing">
        <number>16</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>15</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>8</number>
       </property>
       <item>
        <widget class="QFrame" name="m_pGroupServer">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="styleSheet">
          <string notr="true">.QFrame{
border: 1px solid  rgba(192,192,192, 0.2);;
border-radius: 4px;
background-color: rgba(192,192,192, 0.1);
}</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <property name="topMargin">
           <number>11</number>
          </property>
          <item alignment="Qt::AlignmentFlag::AlignTop">
           <widget class="QRadioButton" name="m_pRadioGroupServer">
            <property name="styleSheet">
             <string notr="true">QRadioButton::indicator{
     top: -7px;
}</string>
            </property>
            <property name="text">
             <string>Use this computer's keyboard and mouse 
(make this computer the &amp;server)</string>
            </property>
           </widget>
          </item>
          <item alignment="Qt::AlignmentFlag::AlignTop">
           <widget class="QWidget" name="m_pWidgetServer" native="true">
            <layout class="QVBoxLayout" name="m_pLayoutServer">
             <property name="spacing">
              <number>10</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QWidget" name="m_pWidgetServerInverse" native="true">
               <layout class="QVBoxLayout" name="m_pLayoutServerInverse">
                <property name="spacing">
                 <number>10</number>
                </property>
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <property name="topMargin">
                 <number>0</number>
                </property>
                <property name="rightMargin">
                 <number>0</number>
                </property>
                <property name="bottomMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QLabel" name="m_pLabelClientIp">
                  <property name="text">
                   <string>Client IP address or hostname:</string>
                  </property>
                  <property name="indent">
                   <number>20</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_3">
                  <property name="leftMargin">
                   <number>20</number>
                  </property>
                  <item>
                   <widget class="QLineEdit" name="m_pLineEditClientIp"/>
                  </item>
                  <item>
                   <widget class="QPushButton" name="m_pButtonConnectToClient">
                    <property name="text">
                     <string>Connect</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="synergy_widgets::ServerStateLabel" name="m_pLabelServerState">
               <property name="text">
                <string>No clients connected</string>
               </property>
               <property name="indent">
                <number>20</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="m_pLabelFingerprint">
               <property name="text">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;TLS enabled (&lt;a href=&quot;#&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#4285f4;&quot;&gt;fingerprint&lt;/span&gt;&lt;/a&gt;)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="textFormat">
                <enum>Qt::TextFormat::RichText</enum>
               </property>
               <property name="indent">
                <number>20</number>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="verticalSpacer_2">
               <property name="orientation">
                <enum>Qt::Orientation::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>10</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <item>
                <spacer name="spacer_2">
                 <property name="orientation">
                  <enum>Qt::Orientation::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item>
                <widget class="QPushButton" name="m_pButtonConfigureServer">
                 <property name="text">
                  <string>&amp;Configure</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="m_pGroupClient">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="styleSheet">
          <string notr="true">.QFrame{
border: 1px solid  rgba(192,192,192, 0.2);
border-radius: 4px;
background-color: rgba(192,192,192, 0.1);
}</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <property name="topMargin">
           <number>11</number>
          </property>
          <item alignment="Qt::AlignmentFlag::AlignTop">
           <widget class="QRadioButton" name="m_pRadioGroupClient">
            <property name="styleSheet">
             <string notr="true">QRadioButton::indicator{
     top: -7px;
}</string>
            </property>
            <property name="text">
             <string>Use another computer’s mouse and keyboard
(make this computer the &amp;client)</string>
            </property>
           </widget>
          </item>
          <item alignment="Qt::AlignmentFlag::AlignTop">
           <widget class="QWidget" name="m_pWidgetClient" native="true">
            <layout class="QVBoxLayout" name="m_pLayoutClient">
             <property name="spacing">
              <number>10</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="m_pLabelServerName">
               <property name="text">
                <string>Server IP address or hostname:</string>
               </property>
               <property name="indent">
                <number>20</number>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <property name="leftMargin">
                <number>20</number>
               </property>
               <item>
                <widget class="QLineEdit" name="m_pLineEditHostname">
                 <property name="minimumSize">
                  <size>
                   <width>0</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="toolTip">
                  <string>Hostname or IP address of the server computer.</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="m_pButtonConnect">
                 <property name="text">
                  <string>Connect</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="synergy_widgets::ClientStateLabel" name="m_pLabelClientState">
               <property name="text">
                <string>Connected to server</string>
               </property>
               <property name="indent">
                <number>20</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>10</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="m_pLabelLogs">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Logs</string>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="verticalSpacer_7">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::Policy::Fixed</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>5</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QFrame" name="m_pGroupLog">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">.QFrame{
border: 1px solid  rgba(192,192,192, 0.2);;
border-radius: 4px;
background-color: rgba(192,192,192, 0.1);
}</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <property name="leftMargin">
        <number>8</number>
       </property>
       <property name="topMargin">
        <number>8</number>
       </property>
       <property name="rightMargin">
        <number>8</number>
       </property>
       <property name="bottomMargin">
        <number>8</number>
       </property>
       <item>
        <widget class="QPlainTextEdit" name="m_pLogOutput">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Courier</family>
           <kerning>true</kerning>
          </font>
         </property>
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
         <property name="undoRedoEnabled">
          <bool>false</bool>
         </property>
         <property name="lineWrapMode">
          <enum>QPlainTextEdit::LineWrapMode::NoWrap</enum>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="maximumBlockCount">
          <number>10000</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>6</number>
      </property>
      <property name="sizeConstraint">
       <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
      </property>
      <property name="topMargin">
       <number>16</number>
      </property>
      <item>
       <widget class="QLabel" name="m_pLabelPadlock">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap resource="../res/Synergy.qrc">:/res/icons/16x16/padlock.png</pixmap>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="m_pStatusLabel">
        <property name="text">
         <string>Ready</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="spacer">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="m_labelNotice">
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">padding: 3px 5px;
border-radius: 3px;
background-color: #3B67D3;
color: #ffffff;
top: 3px;</string>
        </property>
        <property name="text">
         <string notr="true">m_labelNotice</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
        </property>
        <property name="margin">
         <number>0</number>
        </property>
        <property name="openExternalLinks">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Policy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="m_pButtonApply">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>&amp;Apply</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="m_pButtonToggleStart">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>&amp;Start</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <action name="m_pActionAbout">
   <property name="text">
    <string>&amp;About Synergy...</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="m_pActionHelp">
   <property name="text">
    <string>Visit &amp;help site</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="m_pActionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="statusTip">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+Q</string>
   </property>
  </action>
  <action name="m_pActionStartCore">
   <property name="text">
    <string>&amp;Start</string>
   </property>
   <property name="statusTip">
    <string>Run</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="m_pActionStopCore">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>S&amp;top</string>
   </property>
   <property name="statusTip">
    <string>Stop</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="actionShowStatus">
   <property name="text">
    <string>S&amp;how Status</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+H</string>
   </property>
  </action>
  <action name="m_pActionMinimize">
   <property name="text">
    <string>&amp;Hide</string>
   </property>
   <property name="toolTip">
    <string>Hide</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="m_pActionRestore">
   <property name="text">
    <string>&amp;Show</string>
   </property>
   <property name="toolTip">
    <string>Show</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="m_pActionSave">
   <property name="text">
    <string>Save configuration &amp;as...</string>
   </property>
   <property name="statusTip">
    <string>Save the interactively generated server configuration to a file.</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+Alt+S</string>
   </property>
  </action>
  <action name="m_pActionSettings">
   <property name="text">
    <string>Preferences</string>
   </property>
   <property name="toolTip">
    <string>Preferences</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="m_pActivate">
   <property name="text">
    <string>Activate</string>
   </property>
   <property name="toolTip">
    <string>Activate</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>synergy_widgets::ServerStateLabel</class>
   <extends>QLabel</extends>
   <header>widgets/ServerStateLabel.h</header>
  </customwidget>
  <customwidget>
   <class>synergy_widgets::ClientStateLabel</class>
   <extends>QLabel</extends>
   <header>widgets/ClientStateLabel.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../res/Synergy.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>m_pButtonToggleStart</sender>
   <signal>clicked()</signal>
   <receiver>m_pActionStartCore</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>361</x>
     <y>404</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
